jedi/inference/syntax_tree.py

```diff
@@ -695,8 +695,15 @@ def tree_name_to_values(inference_state, context, tree_name):
 
             if expr_stmt.type == "expr_stmt" and expr_stmt.children[1].type == "annassign":
                 correct_scope = parser_utils.get_parent_scope(name) == context.tree_node
-                if correct_scope:
+                ann_assign = expr_stmt.children[1]
+                if correct_scope and name.start_pos < ann_assign.start_pos:
```

Comment:
I just realized that you were probably correct with your solution and the start_pos comparison is not needed, so please remove that again. Sorry.